src/library_exceptions.js

```diff
@@ -102,9 +102,17 @@ var LibraryExceptions = {
     }
   },
 
+  __cxa_init_primary_exception__deps: ['$ExceptionInfo'],
+  __cxa_init_primary_exception: (ptr, type, destructor) => {
+#if EXCEPTION_DEBUG
+    dbg('__cxa_init_primary_exception: ' + [ptrToString(ptr), type, ptrToString(destructor)]);
+#endif
+    return ptr;
+  },
+
   // Here, we throw an exception after recording a couple of values that we need to remember
   // We also remember that it was the last exception thrown as we need to know that later.
-  __cxa_throw__deps: ['$ExceptionInfo', '$exceptionLast', '$uncaughtExceptionCount'],
+  __cxa_throw__deps: ['$exceptionLast', '$uncaughtExceptionCount', '__cxa_init_primary_exception'],
```

Comment:
Why remove `ExceptionInfo` dependency here?  This function clearly directly uses `ExceptionInfo`.

On the other hand I don't see it using `__cxa_init_primary_exception`.

Reply:
This is what's left from my previous (abandoned) approach. Will restore it.